### PR TITLE
Validation service min app version check (EXPOSUREAPP-10956)

### PR DESCRIPTION
**Pull request by mtwalli**
- Check against mock server to see if config is reflected (if needed use Test menu to force update the app config)

## TODO
- [x] Error message 
- [x] Redirect to Google play 

### Ticket 
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-10956

### CODE REVIEW

File: Corona-Warn-App/src/main/res/values-de/dcc_ticketing_strings.xml [hunk 1]
```diff
@@ -8,6 +8,10 @@
     <string name="dcc_ticketing_error_service_provider_error">Es trat ein Fehler bei der Verbindung mit dem Anbieter %1$s auf. Bitte kontaktieren Sie den Anbieter wenn das Problem weiterhin besteht.</string>
     <!-- XTXT: Try again error message-->
     <string name="dcc_ticketing_error_try_again">Es trat ein Fehler bei der Verarbeitung der Daten auf. Bitte versuchen Sie es später erneut.</string>
+    <!-- XTXT: Validation service min version message-->
+    <string name="dcc_ticketing_error_min_version">In der aktuellen Version ihrer Corona-Warn-App wird die Tickerüberprüfung nicht mehr unterstützt. Bitte aktualisieren Sie Ihre App und versuchen Sie die Ticketüberprüfung erneut.</string>
```
Comment (KathrinLuecke) [inline Corona-Warn-App/src/main/res/values-de/dcc_ticketing_strings.xml:12]: Typos: Change to "In der aktuellen Version Ihrer Corona-Warn-App wird die Ticketüberprüfung..."

File: Corona-Warn-App/src/main/res/values-de/dcc_ticketing_strings.xml [hunk 2]
```diff
@@ -54,7 +58,7 @@
     <!-- XTXT: not in AllowList dialog title -->
     <string name="dcc_ticketing_not_in_allow_list_dialog_title">Sicherheitshinweis</string>
     <!-- XTXT: not in AllowList dialog message -->
-    <string name="dcc_ticketing_not_in_allow_list_dialog_message">Der Anbieter ist uns nicht bekannt. Aus Sicherheitsgründen ist die Überprüfung bei Ticketbuchung nur für bekannte Anbieter möglich. Bitte kontaktieren Sie den Anbieter, der ihnen dieses QR-Code bereitgstellt hat.</string>
+    <string name="dcc_ticketing_not_in_allow_list_dialog_message">Der Anbieter ist uns nicht bekannt. Aus Sicherheitsgründen ist die Überprüfung bei Ticketbuchung nur für bekannte Anbieter möglich. Bitte kontaktieren Sie den Anbieter, der ihnen dieses QR-Code bereitgstellt hat und versuchen Sie Ihre App auf die neuste Version zu aktualisieren.</string>
```
Comment (KathrinLuecke) [inline Corona-Warn-App/src/main/res/values-de/dcc_ticketing_strings.xml:61]: Typo: der ihnen dieses...  --> der Ihnen dieses...

**Review by KathrinLuecke — CHANGES_REQUESTED** [review]
Please correct typos

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/qrcode/ui/QrCodeScannerFragment.kt [hunk 3]
```diff
@@ -198,6 +203,19 @@ class QrCodeScannerFragment : Fragment(R.layout.fragment_qrcode_scanner), AutoIn
...
+    private fun showValidationServiceMinVersionDialog(errorMsg: LazyString) {
+        val msg = errorMsg.get(requireContext())
+        val dialogType = DccTicketingDialogType.ErrorDialog(
+            msg = msg,
+            negativeButtonRes = R.string.dcc_ticketing_error_min_version_google_play
+        )
+        dialogType.show(
+            this,
+            positiveButtonAction = { startDecode() },
+            negativeButtonAction = { requireContext().openGooglePlay() }
```
Comment (chiljamgossow) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/qrcode/ui/QrCodeScannerFragment.kt:215]: I could not find the requirement to redirect. Is it from a chat or so?

**Review by KathrinLuecke — CHANGES_REQUESTED** [review]
please correct typos

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/qrcode/ui/QrCodeScannerFragment.kt (hunk 3 above)
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/qrcode/ui/QrCodeScannerFragment.kt:215]: See https://www.figma.com/file/KuDEcMSREg3VhZRhC3FXTY/COVID19_Master_2.14?node-id=40933%3A26462

**Comment by mtwalli** [comment]
> Please correct typos

@KathrinLuecke  Fixed

File: Corona-Warn-App/src/main/res/values-de/dcc_ticketing_strings.xml (hunk 2 above)
Comment (chiljamgossow) [reply Corona-Warn-App/src/main/res/values-de/dcc_ticketing_strings.xml]: Ihnen must be upper case

**Review by BMItr — APPROVED** [review]
code lgtm!
tested on oneplus 8 device

**Review by chiljamgossow — APPROVED** [review]

**Review by KathrinLuecke — APPROVED** [review]
UA approved

**Pull request merged**